lib/matplotlib/tests/test_cache.py

```diff
@@ -0,0 +1,114 @@
+    import mock
+
+import os
+
+from matplotlib import cbook
+from matplotlib.testing._conversion_cache import _ConversionCache, _CacheError
+import pytest
+
+
+def test_cache_basic(tmpdir):
+    def intmp(f):
+        return tmpdir.join(f)
+
+    def fname(f):
+        return str(intmp(f))
+
+    try:
+        cache = _ConversionCache(fname('cache'))
+        intmp('fake.pdf').write_binary(b'this is a fake pdf file')
+        intmp('fake.svg').write_binary(b'this pretends to be an svg file')
+        assert not cache.get(fname('fake.pdf'), fname('fakepdf.png'))
+        assert not cache.get(fname('fake.svg'), fname('fakesvg.png'))
+        assert cache.report() == \
+            {'gets': {fname('fake.pdf'), fname('fake.svg')},
+             'hits': set()}
+
+        intmp('fakepdf.png').write_binary(b'generated from the pdf file')
+        cache.put(fname('fake.pdf'), fname('fakepdf.png'))
+        assert cache.get(fname('fake.pdf'), fname('copypdf.png'))
+        assert intmp('copypdf.png').read() == 'generated from the pdf file'
+        assert cache.report() == \
+            {'gets': {fname('fake.pdf'), fname('fake.svg')},
+             'hits': {fname('fake.pdf')}}
+
+        intmp('fakesvg.png').write_binary(b'generated from the svg file')
+        cache.put(fname('fake.svg'), fname('fakesvg.png'))
+        assert cache.get(fname('fake.svg'), fname('copysvg.png'))
+        assert intmp('copysvg.png').read() == 'generated from the svg file'
+        assert cache.report() == \
+            {'gets': {fname('fake.pdf'), fname('fake.svg')},
+             'hits': {fname('fake.pdf'), fname('fake.svg')}}
+    finally:
+        tmpdir.remove(rec=1)
+
+
+def test_cache_expire(tmpdir):
+    def intmp(*f):
+        return tmpdir.join(*f)
+
+    def fname(*f):
+        return str(intmp(*f))
+
+    try:
+        cache = _ConversionCache(fname('cache'), 10)
+        for i in range(5):
+            pngfile = intmp('cache', 'file%d.png' % i)
+            pngfile.write_binary(b'1234')
+            os.utime(str(pngfile), (i*1000, i*1000))
+
+        cache.expire()
+        assert not os.path.exists(fname('cache', 'file0.png'))
+        assert not os.path.exists(fname('cache', 'file1.png'))
+        assert not os.path.exists(fname('cache', 'file2.png'))
+        assert os.path.exists(fname('cache', 'file3.png'))
+        assert os.path.exists(fname('cache', 'file4.png'))
+
+        intmp('cache', 'onemore.png').write_binary(b'x' * 11)
+        os.utime(fname('cache', 'onemore.png'), (5000, 5000))
+
+        cache.expire()
+        assert not os.path.exists(fname('cache', 'file0.png'))
+        assert not os.path.exists(fname('cache', 'file1.png'))
+        assert not os.path.exists(fname('cache', 'file2.png'))
+        assert not os.path.exists(fname('cache', 'file3.png'))
+        assert not os.path.exists(fname('cache', 'file4.png'))
+        assert not os.path.exists(fname('cache', 'onemore.png'))
+
+    finally:
+        tmpdir.remove(rec=1)
```

Comment:
I didn't see in the documentation that it promises to do any cleanup. But it seems that the [base directory](http://doc.pytest.org/en/latest/tmpdir.html#the-default-base-temporary-directory) gets cleaned up after three test runs.